### PR TITLE
MediMate Frontend Submission.

### PR DESCRIPTION
**Pull request by mailsg**
Hi,

This PR is raised to showcase the implementation of Front end part of the two part app `MediMate`. 

Below are the pointers covered :
- The user logs in to the website, only by typing the email address and password.

- In the **Navigation panel**, the user can see links to:
  - "Doctors."
  - "Reserve" form.
  - "My reservations".
  - "Add doctors" (the link is visible to everybody).
  - "Delete doctors" (the link is visible to everybody).

**Homepage**
- On the main page, the user can see a list of doctors.
- When the user selects a specific item, they can see the details page with its full description.
- In the details page, the user can click the "Reserve" button.
- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.
- Made the app responsive, creating both mobile and desktop versions.

**Delete**
- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
- When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city.
- The user can also access the "Reserve" page from the navigation panel.

**Booking**
- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations.

Kindly review the PR and provide feedback on the same.

Thanks.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 import './App.css';
+import SignUp from './components/auth_pages/Signup';
+import Login from './components/auth_pages/Login';
+import MyReservations from './components/MyReservations';
+import AddDoctor from './components/AddDoctor';
+import DeleteDoctor from './components/DeleteDoctor';
+import Root from './components/Root';
+import ReservationForm from './components/ReservationForm';
+
+const router = createBrowserRouter(
```
Comment (Meltrust) [inline src/App.js:20]: - [ ] For some reason, I can't fetch the doctors, even after logging in:

image

Kindly, make it so the doctors can be listed.  

This means "delete doctor" and reservations don't work either.  Kindly take a look.

If this is not a bug, this means your frontend app needs special extra procedures to be correctly run.  If that is the case, kindly add the specific instructions to your readme. 👍 

_This was tested both on Chrome and Firefox._

Comment (Meltrust) [inline src/App.js:20]: - [ ] When trying to create a doctor, I cannot determine the correct way/format to add the dates, so I always get this error:

image


image



Kindly, include some date selector or include a note in the frontend with the correct format (like mm-dd-yyyy).

That way, your users will be able to fill the form easily, and the UX will be vastly improved. 👍

File: src/App.js [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
+        <Route path="/reservation" element={<ReservationForm />} />
+        <Route path="/reservations" element={<MyReservations />} />
+      </Route>
+      <Route path="/log_in" element={<Login />} />
+      <Route path="/sign_up" element={<SignUp />} />
+    </Route>,
+  ),
+);
 
 function App() {
```
Comment (Meltrust) [inline src/App.js]: - [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed (I found 5):

image

Shouldn't take you more than 5-10 min so don't worry.  

That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
@mailsg

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js [hunk 3]
```diff
@@ -1,29 +1,40 @@
...
+        <Route path="/reservation" element={<ReservationForm />} />
+        <Route path="/reservations" element={<MyReservations />} />
+      </Route>
+      <Route path="/log_in" element={<Login />} />
+      <Route path="/sign_up" element={<SignUp />} />
+    </Route>,
+  ),
+);
+
+const App = () => <RouterProvider router={router} />;
```
Comment (Meltrust) [inline src/App.js:38]: - [ ] Whenever I reload the page so the new doctor I just added shows up in the home section, this happens:

image

The frontend starts an unstoppable loop of requests to the server.  The image shows 2867 requests, but I stopped it by logging out and logging in again.

This must be a UseEffect hook, most likely this one kindly take a look:

image

The same happens in "reservations" and "delete doctor" when reloading the page.

Comment (Meltrust) [inline src/App.js:38]: - [ ] When I try to check my appointments, they don't show up even though they exist in the database:

image

image

It seems I'm not authorized to do so.

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @mailsg,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!










#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js (hunk 3 above)
Comment (forLoop94) [reply src/App.js:38]: @Meltrust, thank you for the timely review. Please try refreshing the page anytime a component does not render properly, maybe the backend is latent in responding.

**Review by Meltrust — APPROVED** [review]
#### Hi @mailsg,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause




Cheers and Happy coding!👏👏👏


------

**Pull request merged**